README.md

```diff
@@ -14,58 +14,252 @@ to your list of dependencies in `mix.exs`:
 ```elixir
 def deps do
   [
-    {:redis_mutex, "~> 0.4.0"}
+    {:redis_mutex, "~> 1.0"}
   ]
 end
 ```
 
+## Upgrading from version 0.X to version 1.X
+
+Version 1.0.0 of `RedisMutex` changes how `RedisMutex` is started and how `with_lock` is used. Key changes include:
+
+1. `RedisMutex` no longer runs as its own application.
+   1. If you need or want to set up a Redix connection specifically for `RedisMutex`, it must be added to your application's supervision tree.
+   2. If you want to re-use an existing Redis connection via Redix, it does not need adding to your application's supervision tree.
+2. Configuration for `RedisMutex` (in your application's `config` files) has changed.
+3. Using `RedisMutex`'s `with_lock` is no longer done via `use RedisMutex`. Instead, your application must call
+   the function `RedisMutex.with_lock/3`.
+4. The code you want to execute in `RedisMutex.with_lock/3` is passed in a zero-arity function instead of in a `do` 
```

Comment:
I woudl show an example of before and after?

Reply:
Yes, a before and after might help here. Thanks for the suggestion.